tests/test_custom_checkers/test_missing_return_statements.py

```diff
@@ -348,3 +349,92 @@ def division_missing_return(x, y) -> int:
             ),
         ):
             self.checker.visit_functiondef(func_node)
+
+
+class TestMissingReturnCheckerZ3Option(pylint.testutils.CheckerTestCase):
+    CHECKER_CLASS = InconsistentReturnChecker
+
+    def setup_method(self) -> None:
+        super().setup_method()
+        self.linter.config.z3 = True
+
+    def test_z3_unfeasible_missing_return(self):
+        src = """
+        def unfeasible_missing_return(x: str) -> str:
+            '''
+            Preconditions:
+                - x[0:2] == "ab"
+            '''
+            if x in "cd":
+                print("empty string")
+            else:
+                return x
+        """
+        z3v = Z3Visitor()
+        mod = z3v.visitor.visit(astroid.parse(src))
+        mod.accept(CFGVisitor())
+        func_node = next(mod.nodes_of_class(nodes.FunctionDef))
+
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(func_node)
+
+    def test_z3_partially_feasible_missing_return(self):
+        src = """
+        def feasible_missing_return(x: int) -> int:
+            '''
+            Preconditions:
+                - x in [1, 2, 3, 4, 5]
+            '''
+            if x > 5:
+                print(x)
+            print(x)
+        """
+        z3v = Z3Visitor()
+        mod = z3v.visitor.visit(astroid.parse(src))
+        mod.accept(CFGVisitor())
+        func_node = next(mod.nodes_of_class(nodes.FunctionDef))
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id="missing-return-statement",
+                node=func_node,
+                args="feasible_missing_return",
+                line=9,
```

Comment:
use `ignore_position` for these tests (throughout)

Reply:
Hi Professor, just to make sure that are we only using `ignore_position` for all z3-related test cases or all test cases. The z3-related tests should only focus on z3 functionality, so position of the error should not matter. However, I guess we do want to check for error position for other tests as we are using custom rendering for this checker.

Reply:
Hi @Raine-Yang-UofT, yes that's a good point. For the checks that use custom rendering, it makes sense to keep the position information; but for the checks that don't, I want to standardize on _not_ including the location information in the test assertion, as that makes the tests more brittle in general.